### PR TITLE
🐛 Fix cert rotation tests

### PR DESCRIPTION
**Pull request by Sunnatillo**
This PR fixes the cert rotation tests changing workflow of the test. 
Previously we checked ironic containers restart when we delete the certificate secret.
With new workflow ironic is updating secret mount without restarting the containers. Now in cert rotation tests we just check if the all ironic pod containers are running for certain period of time without any issue. 

TODO. As a next step we can implement checking querying ironic endpoint with new certificate to confirm  it is working.

### CODE REVIEW

File: test/e2e/cert_rotation.go [hunk 1]
```diff
@@ -72,40 +57,33 @@ func certRotation(ctx context.Context, inputGetter func() CertRotationInput) {
 	}
 	for _, secretName := range secretList {
 		err := clientSet.CoreV1().Secrets(ironicNamespace).Delete(ctx, secretName, metav1.DeleteOptions{})
```
Comment (mboukhalfa) [inline test/e2e/cert_rotation.go:59]: Probably we should add step to confirm the secrets do not exist before checking they are recreated

File: test/e2e/cert_rotation.go [hunk 2]
```diff
@@ -72,40 +57,33 @@ func certRotation(ctx context.Context, inputGetter func() CertRotationInput) {
...
+		return errors.New("secret is being recreated")
 	}, input.E2EConfig.GetIntervals(input.SpecName, "wait-pod-restart")...).Should(BeNil())
+
+	// TODO(Sunnatillo): Further extend the test with querying ironic endpoint with new certificates
+	By("Check if all containers are running in ironic pod")
+	Consistently(func() error {
+		ironicPod, err := getPodFromDeployment(ctx, clientSet, ironicDeployment, ironicNamespace)
+		Expect(err).ToNot(HaveOccurred(), "cannot get ironic from Deployment")
+
+		if areAllContainersRunning(ironicPod) {
```
Comment (mboukhalfa) [inline test/e2e/cert_rotation.go:82]: I think we can check directly if the deployment is running ?

Comment (Sunnatillo) [reply test/e2e/cert_rotation.go:82]: Deployment can show running even though not all containers are running in a pod. I think we should check all the containers.

File: test/e2e/cert_rotation.go (hunk 1 above)
Comment (Sunnatillo) [reply test/e2e/cert_rotation.go:60]: Done

File: test/e2e/cert_rotation.go [hunk 3]
```diff
@@ -72,40 +58,38 @@ func certRotation(ctx context.Context, inputGetter func() CertRotationInput) {
...
-
-	By("Wait for containers in the ironic pod to be restarted")
+	By("Check if the secrets are deleted")
+	for _, secretName := range secretList {
+		_, err := clientSet.CoreV1().Secrets(ironicNamespace).Get(ctx, secretName, metav1.GetOptions{})
+		Expect(err).To(HaveOccurred(), secretName)
+		Expect(apierrors.IsNotFound(err)).To(BeTrue())
+	}
+	By("Wait until secrets are recreated")
 	Eventually(func() error {
```
Comment (adilGhaffarDev) [inline test/e2e/cert_rotation.go:70]:
```suggestion
	Eventually(func(g Gomega) {
```

File: test/e2e/cert_rotation.go [hunk 4]
```diff
@@ -72,40 +58,38 @@ func certRotation(ctx context.Context, inputGetter func() CertRotationInput) {
...
-			for _, ironicContainer := range ironicPod.Status.ContainerStatuses {
-				if ironicContainer.Name == container {
-					notFound = false
-					break
-				}
-			}
-			if notFound {
-				return fmt.Errorf("%s container does not exist in Ironic pod", container)
-			}
+		ironicSecretCreated := isSecretRecreated(ctx, clientSet, ironicNamespace, secretList[0])
```
Comment (adilGhaffarDev) [inline test/e2e/cert_rotation.go:71]: I don't think function is needed here, it's just one line.  
```suggestion
	_, err := clientset.CoreV1().Secrets(namespace).Get(ctx, secretList[0], metav1.GetOptions{})
	Expect(err).ToNot(HaveOccurred(), "secret is being recreated")
```

File: test/e2e/cert_rotation.go [hunk 5]
```diff
@@ -72,40 +58,38 @@ func certRotation(ctx context.Context, inputGetter func() CertRotationInput) {
...
+		mariaDBSecretCreated := true
+		if mariadbEnabled {
+			mariaDBSecretCreated = isSecretRecreated(ctx, clientSet, ironicNamespace, secretList[1])
```
Comment (adilGhaffarDev) [inline test/e2e/cert_rotation.go:74]: same here:
_, err := clientset.CoreV1().Secrets(namespace).Get(ctx, secretList[1], metav1.GetOptions{})
Expect(err).ToNot(HaveOccurred(), "secret is being recreated")

File: test/e2e/cert_rotation.go [hunk 6]
```diff
@@ -72,40 +58,38 @@ func certRotation(ctx context.Context, inputGetter func() CertRotationInput) {
...
-						return fmt.Errorf("%s is not restarted", container.Name)
-					}
-				}
-			}
+		if ironicSecretCreated && mariaDBSecretCreated {
 			return nil
 		}
-		return errors.New("ironic pod is not in running state")
+		return errors.New("secret is being recreated")
 	}, input.E2EConfig.GetIntervals(input.SpecName, "wait-pod-restart")...).Should(BeNil())
```
Comment (adilGhaffarDev) [inline test/e2e/cert_rotation.go:80]:
```suggestion
	}, input.E2EConfig.GetIntervals(input.SpecName, "wait-pod-restart")...).Should(Succeed())
```

Comment (Sunnatillo) [reply test/e2e/cert_rotation.go]: done

File: test/e2e/cert_rotation.go (hunk 5 above)
Comment (Sunnatillo) [reply test/e2e/cert_rotation.go]: done

File: test/e2e/cert_rotation.go (hunk 4 above)
Comment (Sunnatillo) [reply test/e2e/cert_rotation.go]: done

File: test/e2e/cert_rotation.go (hunk 3 above)
Comment (Sunnatillo) [reply test/e2e/cert_rotation.go]: done

File: test/e2e/cert_rotation.go [hunk 7]
```diff
@@ -72,40 +58,35 @@ func certRotation(ctx context.Context, inputGetter func() CertRotationInput) {
...
+	By("Check if the secrets are deleted")
+	for _, secretName := range secretList {
+		_, err := clientSet.CoreV1().Secrets(ironicNamespace).Get(ctx, secretName, metav1.GetOptions{})
+		Expect(err).To(HaveOccurred(), secretName)
+		Expect(apierrors.IsNotFound(err)).To(BeTrue())
 	}
+	By("Wait until secrets are recreated")
+	Eventually(func(g Gomega) {
+		_, err := clientSet.CoreV1().Secrets(ironicNamespace).Get(ctx, secretList[0], metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
```
Comment (adilGhaffarDev) [inline test/e2e/cert_rotation.go:72]:
```suggestion
		g.Expect(err).ToNot(HaveOccurred())
```

File: test/e2e/cert_rotation.go [hunk 8]
```diff
@@ -72,40 +58,35 @@ func certRotation(ctx context.Context, inputGetter func() CertRotationInput) {
...
+		if mariadbEnabled {
+			_, err := clientSet.CoreV1().Secrets(ironicNamespace).Get(ctx, secretList[1], metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
```
Comment (adilGhaffarDev) [inline test/e2e/cert_rotation.go:75]: same here
```suggestion
			g.Expect(err).ToNot(HaveOccurred())
```

File: test/e2e/cert_rotation.go [hunk 9]
```diff
@@ -72,40 +58,35 @@ func certRotation(ctx context.Context, inputGetter func() CertRotationInput) {
...
+		Expect(err).ToNot(HaveOccurred(), "cannot delete this secret: %s", secretName)
+	}
+	By("Check if the secrets are deleted")
+	for _, secretName := range secretList {
+		_, err := clientSet.CoreV1().Secrets(ironicNamespace).Get(ctx, secretName, metav1.GetOptions{})
+		Expect(err).To(HaveOccurred(), secretName)
+		Expect(apierrors.IsNotFound(err)).To(BeTrue())
 	}
+	By("Wait until secrets are recreated")
+	Eventually(func(g Gomega) {
```
Comment (tuminoid) [inline test/e2e/cert_rotation.go:70]: linter nit

**Pull request merged**